arangod/Aql/RestAqlHandler.cpp

```diff
@@ -608,10 +608,9 @@ RestStatus RestAqlHandler::handleUseQuery(std::string const& operation, Query* q
                                           VPackSlice const querySlice) {
   auto closeGuard = scopeGuard([this] { _queryRegistry->close(&_vocbase, _qId); });
 
-  auto self = shared_from_this();
   std::shared_ptr<SharedQueryState> ss = query->sharedState();
```

Comment:
this could be stored in the handler itself to really make use of the SSO

Reply:
You mean the `SharedQueryState` should be stored as a member of the RestAqlHandler?

Reply:
Yes